File: backend/app/models/brief.py
"""Shared brief and summary schemas."""

from __future__ import annotations

from typing import List, Literal

from pydantic import BaseModel, Field


class DocumentReference(BaseModel):
    id: str = Field(default_factory=str)
    name: str = Field(..., min_length=1)
    url: str | None = None
    notes: str | None = None


class SummaryModel(BaseModel):
    project_title: str = Field(default="Untitled Project")
    problem: str | None = None
    solution: str | None = None
    target_users: List[str] = Field(default_factory=list)
    success_metrics: List[str] = Field(default_factory=list)
    constraints: List[str] = Field(default_factory=list)
    timeline: str | None = None
    resources: List[str] = Field(default_factory=list)
    documents: List[str] = Field(default_factory=list)
    opportunity_areas: List[str] = Field(default_factory=list)


class BriefModel(BaseModel):
    project_title: str
    project_description: str
    purpose: str
    expected_outcomes: List[str]
    business_model: List[str]
    constraints: List[str]
    timeline: str
    target_users: List[str]
    documents: List[str]
    opportunity_areas: List[str]
    suggested_reads: List[str]
    ideas_board: List[str]
    success_metrics: List[str]


class AgentRunModel(BaseModel):
    summary: SummaryModel
    brief: BriefModel
    follow_up_questions: List[str]
    thread_id: str


class ConversationTurn(BaseModel):
    role: Literal["user", "assistant", "system"] = "user"
    content: str

File: frontend/src/types/brief.ts
export interface SummaryPayload {
  project_title: string
  problem?: string | null
  solution?: string | null
  target_users: string[]
  success_metrics: string[]
  constraints: string[]
  timeline?: string | null
  resources: string[]
  documents: string[]
  opportunity_areas: string[]
}

export interface LovableBrief {
  project_title: string
  project_description: string
  purpose: string
  expected_outcomes: string[]
  business_model: string[]
  constraints: string[]
  timeline: string
  target_users: string[]
  documents: string[]
  opportunity_areas: string[]
  suggested_reads: string[]
  ideas_board: string[]
  success_metrics: string[]
}

export interface BriefPayload {
  summary: SummaryPayload
  brief: LovableBrief
  follow_up_questions: string[]
  thread_id: string
  assistant_message: string
}

export interface ConversationTurn {
  role: 'user' | 'assistant' | 'system'
  content: string
}

export interface DocumentReference {
  id: string
  name: string
  url?: string | null
  notes?: string | null
}

export interface BriefRunRequest {
  conversation?: ConversationTurn[]
  documents?: DocumentReference[]
  prompt?: string
  thread_id?: string
}
